### PR TITLE
add toOne relationship to dirtyAttributes

### PR DESCRIPTION
**Pull request by jondayton**

### CODE REVIEW

File: spec/Model.spec.js [hunk 1]
```diff
@@ -591,8 +617,22 @@ describe('Model', () => {
...
+        id: 11,
+        description: 'Example description'
+      })
+
+      note.todo = todo
+      note.setPreviousSnapshot()
+      expect(note.dirtyAttributes).toEqual([])
+      note.todo = null
+      expect(note.dirtyAttributes).toEqual(['relationships.todo'])
+    })
```
Comment (jondayton) [inline spec/Model.spec.js:632]: test

File: spec/Model.spec.js [hunk 2]
```diff
@@ -603,6 +643,18 @@ describe('Model', () => {
...
+      const note = store.add('notes', {
+        id: 11,
+        description: 'Example description'
+      })
+
+      expect(note.dirtyAttributes).toEqual([])
+      note.todo = todo
+      expect(note.dirtyAttributes).toEqual(['relationships.todo'])
+    })
+
```
Comment (jondayton) [inline spec/Model.spec.js:657]: test

File: dist/mobx-async-store.cjs.js [hunk 3]
```diff
@@ -2735,13 +2736,17 @@ function setRelatedRecord(record, relatedRecord, property) {
 
   if (!relatedRecord) {
     delete relationships[relationType];
+
+    _dirtyRelationships.add(relationType);
```
Comment (jondayton) [inline dist/mobx-async-store.cjs.js:2740]: code

File: src/decorators/relationships.js [hunk 4]
```diff
@@ -185,8 +185,10 @@ export function setRelatedRecord (record, relatedRecord, property, modelType = n
 
   if (!relatedRecord) {
     delete relationships[relationType]
+    _dirtyRelationships.add(relationType)
   } else if (!data || !(data.type === type && data.id === id)) {
     relationships[relationType] = { data: { id, type } }
+    _dirtyRelationships.add(relationType)
```
Comment (jackiekircher) [inline src/decorators/relationships.js:191]: this case covers when the relatedRecord is changed, yes? There isn't a test for that use case but the attribute should be marked as dirty.

Comment (jondayton) [reply src/decorators/relationships.js:191]: this is for creating a relationship. The other is for updating it.

**Review by jackiekircher — APPROVED** [review]

**Review by jackiekircher — APPROVED** [review]

**Pull request merged**